src/pages/index.js

```diff
@@ -49,35 +65,115 @@ function handleImageClick(data) {
 }
 
 function handleEditSumbit(obj) {
+  popupEdit.renderLoading(true);
   userInfo.setUserInfo(obj);
-  popupEdit.close();
-  formEditValidator.disableButton();
+  api
+    .updateCurrentUser(
+      JSON.stringify({
+        name: obj.name,
+        about: obj.about,
+      })
+    )
+    .then(() => {
+      popupEdit.close();
+      popupEdit.renderLoading(false);
+      formEditValidator.disableButton();
+    });
 }
 
 function handleAddSumbit(obj) {
-  sectionCard.addItem(createCard(obj));
-  popupAdd.close();
-  popupAdd.form.reset();
-  formAddValidator.disableButton();
+  popupAdd.renderLoading(true);
+  api
+    .postCard(
+      JSON.stringify({
+        name: obj.name,
+        link: obj.link,
+      })
+    )
+    .then((data) => {
+      sectionCard.addItem(data, "prepend");
+    })
+    .then(() => {
+      setTimeout(10000);
+      popupAdd.close();
+      popupAdd.renderLoading(false);
+      popupAdd.form.reset();
+      formAddValidator.disableButton();
+    });
+}
+
+function handleDeleteClick(thisCard) {
+  popupDelete.getCardInfo(thisCard);
+  popupDelete.open();
+}
+
+function handleDeleteSubmit(thisCard) {
+  api.deleteCard(thisCard._id);
+  thisCard.deleteCard();
+  popupDelete.close();
+}
+
+function handleAvatarSubmit(obj) {
+  popupAvatar.renderLoading(true);
+  avatar.src = obj.link;
```

Comment:
`  avatar.src = obj.link;
`

You need to update `DOM` only in `then` blocks after a successful response from the server so that when there is an error, it shouldn’t change